### DISCUSSION TITLE
css-extract-rspack-plugin generates empty js files

### DISCUSSION THREAD

**Discussion by edbzn** [web-infra-dev/rspack#6189] [Q&A]
Hi, I wanted to know if there's a way to use the [`RemoveEmptyScriptsPlugin`](https://www.npmjs.com/package/webpack-remove-empty-scripts) with Rspack and the new css-extract-rspack-plugin, so far I configured it but it has no effect, I still get empty js files generated along with the css in the bundle. I also checked if there's an option for that in css-extract-rspack-plugin, but it seems not.
Thanks!

**Comment by JSerFeng** [comment]
Still have empty js file in production mode?

**Reply by JSerFeng** [reply]
Empty js file should not exist in production code, could you give me a repro repo

**Reply by edbzn** [reply]
I'm not able to reproduce with a minimal setup, I'm using a custom Nx executor based on @nx/rspack internal functions, so it might be the reason.

**Comment by edbzn** [comment]
Hi again, so I created a [repository](https://github.com/edbzn/ng-rspack) that replicates the issue. It's not a trivial config as the idea is to support Angular but I hope it can help to find the cause. In `rspack.config.js` the `SuppressExtractedTextChunksWebpackPlugin` is not removing the empty js chunk, I also tried the [webpack-remove-empty-scripts plugin](https://www.npmjs.com/package/webpack-remove-empty-scripts), but I get the same result.

**Reply by edbzn** [reply]
This discussion seems to be a duplicate of https://github.com/web-infra-dev/rspack/issues/3608.
